mongodbatlas/online_archives.go

```diff
@@ -196,10 +196,11 @@ type OnlineArchive struct {
 
 // OnlineArchiveCriteria criteria to use for archiving data.
 type OnlineArchiveCriteria struct {
-	DateField       string  `json:"dateField,omitempty"`
-	DateFormat      string  `json:"dateFormat,omitempty"`
-	ExpireAfterDays float64 `json:"expireAfterDays"`
-	Type            string  `json:"type,omitempty"`
+	DateField       string   `json:"dateField,omitempty"`
+	DateFormat      string   `json:"dateFormat,omitempty"`
+	ExpireAfterDays *float64 `json:"expireAfterDays,omitempty"`
+	Type            string   `json:"type,omitempty"`
+	Query           string   `json:"query,omitempty"`
```

Comment:
hi, couple of things, can we keep the fields sorted? and also add some documentation around the conditions here

```suggestion
	DateField       string   `json:"dateField,omitempty"` // DateField is mandatory when Type is DATE
	DateFormat      string   `json:"dateFormat,omitempty"`
	ExpireAfterDays *float64 `json:"expireAfterDays,omitempty"`
	Query           string `json:"query,omitempty"` // Query is mandatory when Type is CUSTOM
	Type            string   `json:"type,omitempty"`
```

Also not blocking but since we are here I wonder if it would be good to add `partitionFields` which seems like a new field as well